File: samples/shared-element-samples/simple-stack-example-sharedelement-fragment/src/main/java/com/example/fragmenttransitions/core/navigation/BaseKey.java
package com.example.fragmenttransitions.core.navigation;

import android.os.Bundle;
import android.os.Parcelable;

import com.zhuinden.simplestackextensions.fragments.DefaultFragmentKey;

import org.jetbrains.annotations.NotNull;

/**
 * Created by Zhuinden on 2020. 12. 18..
 */

public abstract class BaseKey extends DefaultFragmentKey {
    @NotNull
    @Override
    public String getFragmentTag() {
        return toString();
    }
}

File: samples/advanced-samples/mvvm-sample/src/main/java/com/zhuinden/simplestackexamplemvvm/application/BaseKey.kt
package com.zhuinden.simplestackexamplemvvm.application

import com.zhuinden.simplestack.ServiceBinder
import com.zhuinden.simplestackextensions.fragments.DefaultFragmentKey
import com.zhuinden.simplestackextensions.services.DefaultServiceProvider

abstract class BaseKey : DefaultFragmentKey(), DefaultServiceProvider.HasServices {
    override fun getFragmentTag(): String = toString()
    override fun getScopeTag(): String = fragmentTag

    override fun bindServices(serviceBinder: ServiceBinder) {
    }
}